system/jlib/jutil.cpp

```diff
@@ -1453,6 +1460,17 @@ void StringArray::sortCompare(int (*compare)(const char * const * l, const char
     PARENT::sort(compare);
 }
 
+void StringArray::pruneEmpty()
+{
+    aindex_t idx = ordinality();
+    while(idx > 0)
+    {
+        --idx;
+        if (strlen(PARENT::element(idx))==0)
```

Comment:
Efficiency:  Using strlen is inefficient, much better to check if *element(idx) == 0

Otherwise if there are very long strings in the array you walk all of the characters in the strings, instead of just checking the 1st character.  Not an issue for your usage, but any function in jlib could be used in lots of other situations.